plutus-chain-index/src/Plutus/ChainIndex/Emulator/Handlers.hs

```diff
@@ -82,18 +82,18 @@ handleQuery = \case
     UtxoSetMembership r -> do
         utxoState <- gets (measure . view utxoIndex)
         case tip utxoState of
-            Nothing -> throwError QueryFailedNoTip
-            Just tp -> pure (tp, isUnspentOutput r utxoState)
+            TipAtGenesis -> throwError QueryFailedNoTip
+            tp           -> pure (tp, isUnspentOutput r utxoState)
     UtxoSetAtAddress cred -> do
         state <- get
         let outRefs = view (diskState . addressMap . at cred) state
             utxoState = view (utxoIndex . to measure) state
             page = pageOf def $ Set.filter (\r -> isUnspentOutput r utxoState) (fromMaybe mempty outRefs)
         case tip utxoState of
-            Nothing -> throwError QueryFailedNoTip
-            Just tp -> pure (tp, page)
+            TipAtGenesis -> throwError QueryFailedNoTip
```

Comment:
We could probably return an empty page here and emit a warning ("It appears that the chain index has not been synchronised with the node" or something)

Reply:
In order to emit a warning do I need to add a constructor to the log sum type?

Reply:
Is this what you meant? There is no textual representation for the chain-index log messages, so I just added a sum type.